addons/point_of_sale/models/stock_picking.py

```diff
@@ -0,0 +1,130 @@
+                'name': first_line.name,
+                'product_uom': first_line.product_id.uom_id.id,
+                'picking_id': self.id,
+                'picking_type_id': self.picking_type_id.id,
+                'product_id': first_line.product_id.id,
+                'product_uom_qty': abs(sum(order_lines.mapped('qty'))),
+                'state': 'draft',
+                'location_id': self.location_id.id,
+                'location_dest_id': self.location_dest_id.id,
+                'company_id': self.company_id.id,
+            })
+            if first_line.product_id.tracking != 'none' and (self.picking_type_id.use_existing_lots or self.picking_type_id.use_create_lots):
+                for line in order_lines:
+                    sum_of_lots = 0
+                    for lot in line.pack_lot_ids.filtered(lambda l: l.lot_name):
+                        if line.product_id.tracking == 'serial':
+                            qty = 1
+                        else:
+                            qty = abs(line.qty)
+                        ml_vals = current_move._prepare_move_line_vals()
+                        ml_vals.update({'qty_done':qty})
+                        if self.picking_type_id.use_existing_lots:
+                            existing_lot = self.env['stock.production.lot'].search([
+                                ('company_id', '=', self.company_id.id),
+                                ('product_id', '=', line.product_id.id),
+                                ('name', '=', lot.lot_name)
+                            ])
+                            if not existing_lot and self.picking_type_id.use_create_lots:
+                                existing_lot = self.env['stock.production.lot'].create({
+                                    'company_id': self.company_id.id,
+                                    'product_id': line.product_id.id,
+                                    'name': lot.lot_name,
+                                })
+                            ml_vals.update({
+                                'lot_id': existing_lot.id,
+                            })
+                        else:
+                            ml_vals.update({
+                                'lot_name': lot.lot_name,
+                            })
+                        self.env['stock.move.line'].create(ml_vals)
+                        sum_of_lots += qty
+                    if abs(line.qty) != sum_of_lots:
+                        difference_qty = abs(line.qty) - sum_of_lots
+                        ml_vals = current_move._prepare_move_line_vals()
+                        if line.product_id.tracking == 'serial':
+                            ml_vals.update({'qty_done': 1})
+                            for i in range(int(difference_qty)):
+                                self.env['stock.move.line'].create(ml_vals)
+                        else:
+                            ml_vals.update({'qty_done': 1})
```

Comment:
Not sure if this is correct. I think the `qty_done` is `difference_qty` instead of `1`.